index.php

```diff
@@ -10,18 +10,49 @@ function e($string)
     return htmlspecialchars($string);
 }
 
+/**
+ * @param $amount
+ * @return string
+ */
 function format_amount($amount)
 {
     $amount = ceil($amount);
     if ($amount >= 1000) {
         $amount = number_format($amount, 0, ".", " ");
     }
-
     return "{$amount} ₽";
-    /**еще короче можно
-     * return number_format(ceil($amount),0, '.',' ').' ₽';
-     * но по условию задания мы проверяем if ($amount >= 1000)
-     */
+}
+
+
+/**
+ *
+ * @param $fin_time => 'ГГГГ-ММ-ДД'
+ * @return array['$h' => 'string','$m' => 'string'])]
+ *
+ */
+function countdown($fin_time) //однообразил с ключом массива Units
+{
+    date_default_timezone_set('Europe/Kaliningrad');
+    //про константу не совсем понял, что куда вынести
+    $deadline = strtotime($fin_time);//дата истечения срока
+    $currentTime = time();//текущее время
+    if ($deadline > $currentTime) {
+        $s = $deadline - $currentTime;//осталось секунд до дедлайна
+        $h = floor($s / 60 / 60);//осталось часов
+        $m = floor($s / 60) - ($h * 60);//осталось минут
+        if ($h < 10) {
+            $h = str_pad($h, 2, "0", STR_PAD_LEFT); // Добавляем лидирующий ноль
+        }
+        if ($m < 10) {
+            $m = str_pad($m, 2, "0", STR_PAD_LEFT); // Добавляем лидирующий ноль
+        }
+    } else {
+        $h = $m = '00';//время вышло
+    }
+    settype($h, "string");
```

Comment:
it is interesting way but isn't the best way to do typing. See https://www.php.net/manual/ru/function.settype.php#123490

I recommend to use `return [(string)$h, (string)$m];`. Type casting this way is recommender and commonly used